### PR TITLE
feat: adding loaded future to the component

### PR DESCRIPTION
**Pull request by erickzanardo**
# Description

Adds `loaded` getter on the component, similar to `Component.mounted`.

## Checklist

- [x] The title of my PR starts with a [Conventional Commit] prefix (`fix:`, `feat:`, `docs:` etc).
- [x] I have read the [Contributor Guide] and followed the process outlined for submitting PRs.
- [x] I have updated/added tests for ALL new/updated/fixed functionality.
- [x] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [ ] I have updated/added relevant examples in `examples`.

## Breaking Change

- [ ] Yes, this is a breaking change.
- [x] No, this is *not* a breaking change.

## Related Issues

[issue database]: https://github.com/flame-engine/flame/issues
[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Flame Style Guide]: https://github.com/flame-engine/flame/blob/main/STYLEGUIDE.md
[Conventional Commit]: https://conventionalcommits.org

### CODE REVIEW

**Comment by erickzanardo** [comment]
cc @alestiago

**Review by alestiago — APPROVED** [review]

File: packages/flame_flare/example/lib/main.dart [hunk 1]
```diff
@@ -19,8 +19,6 @@ class MyGame extends FlameGame with TapDetector, DoubleTapDetector {
 
   late final MinionComponent minionComponent;
 
-  bool loaded = false;
```
Comment (erickzanardo) [inline packages/flame_flare/example/lib/main.dart:22]: Note: this wasn't being used and was conflicting with the new getter, so I just removed it

**Review by wolfenrain — APPROVED** [review]

**Review by spydon — APPROVED** [review]
MIssing docs, but other than that it looks good!
I would probably have expected a `bool` from a field named `loaded`, but since we have the same with `mounted` we should follow that.

File: doc/flame/components.md [hunk 2]
```diff
@@ -28,14 +28,19 @@ the `Component` remove method.
...
 throughout the component's lifetime. This method will only run if the parent is already mounted.
 If the parent is not mounted yet, then this method will wait in a queue (this will have no effect
-on the rest of the game engine). 
+on the rest of the game engine).
 
+A component lifecycle state can be checked by a series of getters:
+ - `isLoaded`: Returns a bool with the current loaded state
+ - `loaded`: Retunrs a future that will complete once the component has finished loading
+ - `isMounted`: Returns a bool with the current mounted state
+ - `mounted`: Retunrs a future that will complete once the component has finished mounting
```
Comment (spydon) [inline doc/flame/components.md:43]:
```suggestion
 - `loaded`: Returns a future that will complete once the component has finished loading
 - `isMounted`: Returns a bool with the current mounted state
 - `mounted`: Returns a future that will complete once the component has finished mounting
```

**Review by spydon — APPROVED** [review]

**Pull request merged**